Got "not existing provider" with GHE

version: f6fe36560708d5d59476f570780f7ec59967104c
Go: 1.13
OS: macOS Catalina (10.15.7)

got an error when use `--github-api-url=https://own.ghe.host.domain` option. The host name `own.ghe.host.domain` cannot pass this resolver. https://github.com/google/triage-party/blob/0f7218d3e80239ee9d6e058deca1d4bddf221c5d/pkg/provider/provider.go#L42


Error log:

```
not existing provider
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1779c69]

goroutine 228 [running]:
github.com/google/triage-party/pkg/hubbub.(*Engine).updateIssues(0xc0017f8000, 0x1b6d7a0, 0xc0000a6000, 0xc00166a0a9, 0x4, 0xc00166a0ae, 0xa, 0xc00166a098, 0x10, 0x0, ...)
	/path/to/triage-party/pkg/hubbub/issue.go:87 +0x4d9
github.com/google/triage-party/pkg/hubbub.(*Engine).cachedIssues(0xc0017f8000, 0x1b6d7a0, 0xc0000a6000, 0xc00166a0a9, 0x4, 0xc00166a0ae, 0xa, 0xc00166a098, 0x10, 0x0, ...)
	/path/to/triage-party/pkg/hubbub/issue.go:46 +0x380
github.com/google/triage-party/pkg/hubbub.(*Engine).SearchIssues.func1(0xc00174c100, 0xc001814000, 0xc0017f8000, 0x1b6d7a0, 0xc0000a6000, 0xc001908440, 0xc001908400)
	/path/to/triage-party/pkg/hubbub/search.go:66 +0x16b
created by github.com/google/triage-party/pkg/hubbub.(*Engine).SearchIssues
	/path/to/triage-party/pkg/hubbub/search.go:58 +0x42a
exit status 2
```

--- a/pkg/provider/provider.go
+++ b/pkg/provider/provider.go
@@ -52,6 +52,12 @@ func ResolveProviderByHost(providerHost string) Provider {
 		}
 		return gitlabProvider
 	}
+	// support own host with GitHub Enterprise
+	if githubProvider != nil {
+		if providerHost == githubProvider.client.BaseURL.Host {
+			return githubProvider
+		}
+	}
 	fmt.Println("not existing provider")
 	return nil
 }

